### PR TITLE
🐛 Fix k8s.ingress certificates

### LINKED ISSUE #1867: k8s scan compare results
```
FTL failed to resolve policies error="failed to compile bundle: failed to compile //local.cnspec.io/run/local-execution/queries/mondoo-kubernetes-best-practices-ingress-cert-expiration: failed to compile query 'k8s.ingress.tls.all(\n  certificates.all(\n    expiresIn.days > 15\n  )\n)\n': cannot find resource that is called by '>' of type core.certificate\nfailed to compile //local.cnspec.io/run/local-execution/queries/mondoo-kubernetes-best-practices-ingress-cert-expiration: failed to compile query 'k8s.ingress.tls.all(\n  certificates.all(\n    expiresIn.days > 15\n  )\n)\n': cannot find resource that is called by '>' of type core.certificate\n"
exit status 1
```

### PR DESCRIPTION
**Pull request by czunker**
Fixes #1867

### CODE REVIEW

**Comment by czunker** [comment]
This works now with Ingress objects which have and don't have TLS certificates configured:
```
cnquery run k8s --discover ingresses -c "k8s.ingress.tls.all( certificates.all( expiresIn.days > 15 ) )"
! using builtin provider for k8s
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
! using builtin provider for k8s
[ok] value: true
```

File: providers/k8s/resources/ingress.go [hunk 1]
```diff
@@ -20,8 +20,10 @@ import (
 )
 
 type mqlK8sIngressInternal struct {
-	lock sync.Mutex
-	obj  *networkingv1.Ingress
+	lock   sync.Mutex
+	obj    *networkingv1.Ingress
+	objId  string
+	mqlK8s *mqlK8s
```
Comment (czunker) [inline providers/k8s/resources/ingress.go]: I only need this to later access the `GetSecrets` func. Perhaps there is a nicer way of doing this. Ideas are very welcome.

Comment (arlimus) [reply providers/k8s/resources/ingress.go]: You can request resources on the fly, especially useful if they are static like in `k8s` case:

```go
	o, err := CreateResource(runtime, "k8s", map[string]*llx.RawData{})
	if err != nil {
		return nil, err
	}
	k8s := o.(*mqlK8s)
```

Comment (czunker) [reply providers/k8s/resources/ingress.go]: Thanks for this. Works like a charm.

File: providers/k8s/resources/ingress.go [hunk 2]
```diff
@@ -60,16 +56,33 @@ func (k *mqlK8s) ingresses() ([]interface{}, error) {
...
 
+func (k *mqlK8sIngress) tls() ([]interface{}, error) {
+	o, err := CreateResource(k.MqlRuntime, "k8s", map[string]*llx.RawData{})
+	if err != nil {
+		return nil, err
+	}
+	k8s := o.(*mqlK8s)
+
+	ingress := k.obj
+	objId := k.objId
```
Comment (arlimus) [inline providers/k8s/resources/ingress.go]: nit: since these variables are only used once, you could just apply them

**Review by arlimus — APPROVED** [review]
Looks great, thank you for the update!

File: providers/k8s/resources/ingress.go (hunk 2 above)
Comment (czunker) [reply providers/k8s/resources/ingress.go]: Good point. Thanks.

**Pull request merged**